### DISCUSSION TITLE
is it normal that 5 of the automated tests in justpy fail with errors?

### DISCUSSION THREAD

**Discussion by freemant2000** [elimintz/justpy#676] [General]
I just tried running the tests and got errors:
```
Ran 44 tests in 83.946s using 1 process

FAILED (errors=5, failures=1, passes=33, skips=5)

```
Is this expected?
Thanks!

**Comment by WolfgangFahl** [comment]
@freemant2000 - thank you Kent for bringing this up. The automated tests run a bit differently in the CI enviroment than locally. The success depends a lot on your envionment - especially the Operating System and the Python version. Please let us know which tests fail. I'd assume the problem is with the selenium tests since these need to load drivers that are specific to the browser version being used for the tests.

**Reply by freemant2000** [reply]
Right, the tests in test_restful.py  failed because the python-multipart package was not installed.

Does it mean that it should be listed below as a test package?
```
install_packages() {
  # install test packages
  pip install aiohttp asynctest green selenium  testfixtures webdriver-manager python-multipart
  # install example packages
  pip install bokeh matplotlib pandas plotly pydeck pygments seaborn vega_datasets
}
```

**Reply by freemant2000** [reply]
Now, only test_issue531.py fails as "Bad session" is not found in the response, which is just a regular  justpy page.

**Reply by WolfgangFahl** [reply]
@freemant2000  Great - i think we need to install the python multipart dependency then and look into why test_issue531 fails. The last time the CI ran there was only a timeout issue so we might want to trigger the CI again to see how things go
